### PR TITLE
Process h264d - use C++ primitives: memcpy => std::copy

### PR DESCRIPTION
**Pull request by ilapichev**
Issue: MDP-50956

### CODE REVIEW

**Comment by ilapichev** [comment]
Please take a closer look at umc_media_data.h

Original:
virtual void* GetDataPointer(void)   { return m_pDataPointer; }
Now:
virtual uint8_t* GetDataPointer(void)   { return m_pDataPointer; }

This method was changed because std::copy function does not accept void* as 3 parameter.

**Comment by mgonchar** [comment]
It is already done in https://github.com/Intel-Media-SDK/MediaSDK/pull/724/

File: _studio/mfx_lib/decode/h264/src/mfx_h264_dec_decode.cpp [hunk 1]
```diff
@@ -1640,7 +1640,7 @@ mfxStatus VideoDECODEH264::GetUserData(mfxU8 *ud, mfxU32 *sz, mfxU64 *ts)
     if (!m_isInit)
         return MFX_ERR_NOT_INITIALIZED;
 
-    MFX_CHECK_NULL_PTR3(ud, sz, ts);
+    //MFX_CHECK_NULL_PTR3(ud, sz, ts);
```
Comment (mgonchar) [inline _studio/mfx_lib/decode/h264/src/mfx_h264_dec_decode.cpp]: why you decided to remove check?

File: _studio/mfx_lib/decode/h264/src/mfx_h264_dec_decode.cpp [hunk 2]
```diff
@@ -708,9 +708,9 @@ mfxStatus VideoDECODEH264::GetVideoParam(mfxVideoParam *par)
             return MFX_ERR_NOT_ENOUGH_BUFFER;
         }
 
-        memcpy_s(mvcSeqDesc->View, mvcSeqDesc->NumView * sizeof(mfxMVCViewDependency), mvcSeqDescInternal->View, mvcSeqDescInternal->NumView * sizeof(mfxMVCViewDependency));
-        memcpy_s(mvcSeqDesc->ViewId, mvcSeqDesc->NumViewId * sizeof(mfxU16), mvcSeqDescInternal->ViewId, mvcSeqDescInternal->NumViewId * sizeof(mfxU16));
-        memcpy_s(mvcSeqDesc->OP, mvcSeqDesc->NumOP * sizeof(mfxMVCOperationPoint), mvcSeqDescInternal->OP, mvcSeqDescInternal->NumOP * sizeof(mfxMVCOperationPoint));
+        std::copy(mvcSeqDesc->View, mvcSeqDesc->View + mvcSeqDesc->NumView * sizeof(mfxMVCViewDependency), mvcSeqDescInternal->View);
```
Comment (mgonchar) [inline _studio/mfx_lib/decode/h264/src/mfx_h264_dec_decode.cpp]: std::copy has different parameter order than memcpy:

```
errno_t memcpy_s(  
   void *dest, // <- dest
   size_t destSize,  
   const void *src, // <- src
   size_t count   
);
```
But std::copy has other
```
template< class InputIt, class OutputIt >
OutputIt copy( InputIt first, InputIt last, <- src
OutputIt d_first ); // <- dest
```

File: _studio/shared/umc/core/umc/include/umc_media_data.h [hunk 3]
```diff
@@ -73,7 +73,7 @@ class MediaData
 
     // Get an address of the beginning of valid data.
     // This pointer could not be equal to the beginning of the buffer.
-    virtual void* GetDataPointer(void)   { return m_pDataPointer; }
+    virtual uint8_t* GetDataPointer(void)   { return m_pDataPointer; }
```
Comment (mgonchar) [inline _studio/shared/umc/core/umc/include/umc_media_data.h]: You could workaround it with cast in place

**Comment by ilapichev** [comment]
already done

**Pull request closed without merging**